lib/blockscore/errors/api_error.rb

```diff
@@ -15,19 +17,29 @@ class APIError < Error
     # APIError - Indicates an error on the server side (HTTP 5xx)
     # AuthenticationError - Indicates an authentication error (HTTP 401)
     def initialize(response)
-      body = JSON.parse(response.body, symbolize_names: true)
+      @http_body = JSON.parse(response.body, symbolize_names: true)
 
-      @message = body[:error][:message]
+      @message = error_advice[:message]
       @http_status = response.code
-      @error_type = body[:error][:type]
-      @http_body = body
+      @error_type = error_advice[:type]
     end
 
     def to_s
-      status_string = @http_status ? "(Status: #{@http_status})" : ''
-      type_string = @error_type ? "(Type: #{@error_type})" : ''
-
       "#{type_string} #{@message} #{status_string}"
```

Comment:
I realize this wasn't changed, but in the event `type_string` is nil, do you want the error message to have a leading space? I might be more inclined to write this as:

```ruby
def to_s
  [type_string, @message, status_string].compact.join(' ')
end
```